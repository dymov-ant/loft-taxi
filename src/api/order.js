import { api } from './index';

export const getAddressList = () => api.get('/addressList');

export const getRoute = (payload) =>
  api.get('/route', {
    params: {
      address1: payload.addressFrom,
      address2: payload.addressTo,
    },
  });
