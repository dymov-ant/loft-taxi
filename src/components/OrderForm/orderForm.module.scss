.form {
  &__top {
    margin-bottom: 10px;

    @media screen and (max-width: 767px) {
      margin-bottom: 5px;
    }
  }

  &__tariff {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 30px;

    @media screen and (max-width: 767px) {
      margin-bottom: 10px;
    }
  }
}

.tariffItem {
  display: flex;
  flex-direction: column;
  background: #FFFFFF;
  box-shadow: 0 3px 20px rgba(0, 0, 0, 0.2);
  border-radius: 10px;
  opacity: 0.5;
  cursor: pointer;
  padding: 10px;

  @media screen and (max-width: 767px) {
    padding: 7px;
  }

  &_active {
    opacity: 1;
  }

  &__name {
    font-size: 16px;
    line-height: 19px;
    color: #000000;
    margin-bottom: 8px;
  }

  &__price {
    font-size: 11px;
    line-height: 13px;
    color: #828282;

    span {
      display: block;
      font-size: 24px;
      line-height: 28px;
      color: #000000;
      margin-bottom: 10px;
    }
  }

  img {
    @media screen and (max-width: 767px) {
      display: none;
    }
  }
}

.successForm {
  &__title {
    font-weight: bold;
    font-size: 36px;
    line-height: 42px;
    margin-bottom: 14px;

    @media screen and (max-width: 767px) {
      font-size: 21px;
      line-height: 25px;
      text-align: center;
      margin-bottom: 10px;
    }
  }

  &__text {
    font-size: 18px;
    line-height: 21px;
    color: #7B7B7B;
    margin-bottom: 35px;

    @media screen and (max-width: 767px) {
      font-size: 16px;
      line-height: 19px;
      text-align: center;
      margin-bottom: 15px;
    }
  }
}