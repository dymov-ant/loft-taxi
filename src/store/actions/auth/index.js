import {
  FETCH_LOGIN,
  FETCH_REGISTRATION,
  LOGOUT,
  SET_ERROR,
  SET_IS_LOADING,
  SET_IS_LOGGED_IN,
} from '../../reducers/auth/types';

export const authActions = {
  setIsAuth: (payload) => ({ type: SET_IS_LOGGED_IN, payload }),
  setIsLoading: (payload) => ({ type: SET_IS_LOADING, payload }),
  setError: (payload) => ({ type: SET_ERROR, payload }),
  fetchLogin: (payload) => ({ type: FETCH_LOGIN, payload }),
  fetchRegistration: (payload) => ({ type: FETCH_REGISTRATION, payload }),
  logout: () => ({ type: LOGOUT }),
};
