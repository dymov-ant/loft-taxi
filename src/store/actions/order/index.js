import {
  CLEAR_ROUTE,
  FETCHING_ADDRESS_LIST,
  FETCHING_ERROR_ADDRESS_LIST,
  FETCHING_ERROR_ROUTE,
  FETCHING_ROUTE,
  FETCHING_SUCCESS_ADDRESS_LIST,
  FETCHING_SUCCESS_ROUTE,
  GET_ADDRESS_LIST,
  GET_ROUTE,
} from '../../reducers/order/types';

export const orderActions = {
  fetchingAddressList: () => ({ type: FETCHING_ADDRESS_LIST }),
  fetchingSuccessAddressList: (addressList) => ({ type: FETCHING_SUCCESS_ADDRESS_LIST, payload: addressList }),
  fetchingErrorAddressList: (error) => ({ type: FETCHING_ERROR_ADDRESS_LIST, payload: error }),
  getAddressList: () => ({ type: GET_ADDRESS_LIST }),
  fetchingRoute: () => ({ type: FETCHING_ROUTE }),
  fetchingSuccessRoute: (route) => ({ type: FETCHING_SUCCESS_ROUTE, payload: route }),
  fetchingErrorRoute: (error) => ({ type: FETCHING_ERROR_ROUTE, payload: error }),
  clearRoute: () => ({ type: CLEAR_ROUTE }),
  getRoute: (addressFrom, addressTo) => ({ type: GET_ROUTE, payload: { addressFrom, addressTo } }),
};
