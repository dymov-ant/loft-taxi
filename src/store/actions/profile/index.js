import {
  FETCHING_ERROR_PROFILE,
  FETCHING_PROFILE,
  FETCHING_SUCCESS_PROFILE,
  GET_PROFILE,
  IS_SUCCESS_SUBMITTING,
  SET_PROFILE
} from "../../reducers/profile/types"

export const profileActions = {
  fetchingProfile: () => ({type: FETCHING_PROFILE}),
  fetchingSuccess: (card) => ({type: FETCHING_SUCCESS_PROFILE, payload: card}),
  fetchingError: (error) => ({type: FETCHING_ERROR_PROFILE, payload: error}),
  successSubmitting: (isSubmitting) => ({type: IS_SUCCESS_SUBMITTING, payload: isSubmitting}),
  getProfile: () => ({type: GET_PROFILE}),
  setProfile: (cardData) => ({type: SET_PROFILE, payload: cardData})
}