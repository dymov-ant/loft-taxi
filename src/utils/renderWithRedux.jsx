import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { createMemoryHistory } from 'history';
// eslint-disable-next-line import/no-extraneous-dependencies
import configureStore from 'redux-mock-store';

const history = createMemoryHistory();
const mockStore = configureStore();

export const renderWithRedux = (component, initialState = {}) => {
  const store = mockStore(initialState);
  return render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  );
};
