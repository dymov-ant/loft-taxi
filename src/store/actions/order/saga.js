import { call, put, takeEvery } from "redux-saga/effects"
import { orderActions } from "./index"
import { getAddressList, getRoute } from "../../../api/order"
import { GET_ADDRESS_LIST, GET_ROUTE } from "../../reducers/order/types"

function* getAddressListWorker() {
  try {
    yield put(orderActions.fetchingAddressList())
    const response = yield call(getAddressList)
    if (response.data.addresses.length) {
      yield put(orderActions.fetchingSuccessAddressList(response.data.addresses))
    } else {
      yield put(orderActions.fetchingErrorAddressList("Ошибка при получении списка адресов"))
      console.log("Ошибка при получении списка адресов")
    }
  } catch (e) {
    console.log("Ошибка при запросе:", e)
    yield put(orderActions.fetchingErrorAddressList("Что-то пошло не так"))
  }
}

function* getRouteWorker(action) {
  try {
    yield put(orderActions.fetchingRoute())
    const {payload} = action
    const response = yield call(getRoute, payload)
    console.log(response)
    if (response.data.length) {
      yield put(orderActions.fetchingSuccessRoute(response.data))
    } else {
      yield put(orderActions.fetchingErrorRoute("Ошибка при получении маршрута"))
      console.log("Ошибка при получении маршрута")
    }
  } catch (e) {
    console.log("Ошибка при запросе:", e)
    yield put(orderActions.fetchingErrorRoute("Что-то пошло не так"))
  }
}

export function* orderWatcher() {
  yield takeEvery(GET_ADDRESS_LIST, getAddressListWorker)
  yield takeEvery(GET_ROUTE, getRouteWorker)
}