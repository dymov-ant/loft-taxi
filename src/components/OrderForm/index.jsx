import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import classNames from 'classnames/bind';
import { Button, FormControl, InputLabel, MenuItem, Select } from '@material-ui/core';
import { orderActions } from '../../store/actions/order';
import Spinner from '../Spinner';
import standardImg from '../../assets/img/tariffs/tariff_1.png';
import premiumImg from '../../assets/img/tariffs/tariff_2.png';
import businessImg from '../../assets/img/tariffs/tariff_3.png';
import style from './orderForm.module.scss';

const OrderForm = () => {
  const cx = classNames.bind(style);
  const dispatch = useDispatch();
  const addressList = useSelector((state) => state.order.addressList);
  const isLoading = useSelector((state) => state.order.isLoading);
  const route = useSelector((state) => state.order.route);

  const [addressFrom, setAddressFrom] = useState('');
  const [addressTo, setAddressTo] = useState('');
  const [tariff, setTariff] = useState(1);

  const handleSelect = (event) => {
    if (event.target.name === 'from') {
      setAddressFrom(event.target.value);
    } else {
      setAddressTo(event.target.value);
    }
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    dispatch(orderActions.getRoute(addressFrom, addressTo));
  };

  const handleReset = () => {
    setTariff(1);
    setAddressFrom('');
    setAddressTo('');
    dispatch(orderActions.clearRoute());
  };

  return route.length > 0 ? (
    <div className={style.successForm}>
      <p className={style.successForm__title}>Заказ размещен</p>
      <p className={style.successForm__text}>Ваше такси уже едет к вам. Прибудет приблизительно через 10 минут.</p>
      <Button color="primary" fullWidth variant="contained" onClick={handleReset}>
        Сделать новый заказ
      </Button>
    </div>
  ) : (
    <div className={style.form}>
      <div className={style.form__top}>
        <FormControl fullWidth margin="normal" disabled={isLoading}>
          <InputLabel>Откуда</InputLabel>
          <Select
            name="from"
            value={addressFrom}
            onChange={handleSelect}
            MenuProps={{
              anchorOrigin: {
                vertical: 'bottom',
                horizontal: 'left',
              },
              getContentAnchorEl: null,
            }}
          >
            {addressList.map((item) => (
              <MenuItem value={item} key={item}>
                {item}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
        <FormControl fullWidth margin="normal" disabled={!addressFrom}>
          <InputLabel>Куда</InputLabel>
          <Select
            name="to"
            value={addressTo}
            onChange={handleSelect}
            MenuProps={{
              anchorOrigin: {
                vertical: 'bottom',
                horizontal: 'left',
              },
              getContentAnchorEl: null,
            }}
          >
            {addressList.map(
              (item) =>
                item !== addressFrom && (
                  <MenuItem value={item} key={item}>
                    {item}
                  </MenuItem>
                ),
            )}
          </Select>
        </FormControl>
      </div>
      {addressFrom && addressTo && (
        <div className={style.form__tariff}>
          <div
            className={cx(style.tariffItem, { [style.tariffItem_active]: tariff === 1 })}
            onClick={() => setTariff(1)}
            data-testid="tariff-1"
          >
            <p className={style.tariffItem__name}>Стандарт</p>
            <p className={style.tariffItem__price}>
              Стоимость <span>150 ₽</span>
            </p>
            <img src={standardImg} alt="стандарт" />
          </div>
          <div
            className={cx(style.tariffItem, { [style.tariffItem_active]: tariff === 2 })}
            onClick={() => setTariff(2)}
            data-testid="tariff-2"
          >
            <p className={style.tariffItem__name}>Премиум</p>
            <p className={style.tariffItem__price}>
              Стоимость <span>250 ₽</span>
            </p>
            <img src={premiumImg} alt="стандарт" />
          </div>
          <div
            className={cx(style.tariffItem, { [style.tariffItem_active]: tariff === 3 })}
            onClick={() => setTariff(3)}
            data-testid="tariff-3"
          >
            <p className={style.tariffItem__name}>Бизнес</p>
            <p className={style.tariffItem__price}>
              Стоимость <span>300 ₽</span>
            </p>
            <img src={businessImg} alt="стандарт" />
          </div>
        </div>
      )}
      {isLoading ? (
        <Spinner />
      ) : (
        <Button
          color="primary"
          fullWidth
          variant="contained"
          onClick={handleSubmit}
          disabled={!(addressTo && addressFrom && tariff)}
        >
          Заказать
        </Button>
      )}
    </div>
  );
};

export default OrderForm;
