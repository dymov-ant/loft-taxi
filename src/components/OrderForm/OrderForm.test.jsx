import React from 'react';
import { fireEvent, within } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import OrderForm from './index';
import { renderWithRedux } from '../../utils/renderWithRedux';
import { orderActions } from '../../store/actions/order';

jest.mock('../Spinner', () => () => <div>spinner</div>);

describe('OrderForm component', () => {
  let initialState;

  beforeEach(() => {
    initialState = {
      order: {
        addressList: ['address1', 'address2', 'address3'],
        route: [],
        isLoading: false,
      },
    };
  });

  it('OrderForm renders', () => {
    const { container } = renderWithRedux(<OrderForm />, initialState);
    expect(container).toBeInTheDocument();
  });

  it('OrderForm handleSelect', () => {
    const { getByRole, getAllByRole } = renderWithRedux(<OrderForm />, initialState);
    const [fromSelect, toSelect] = getAllByRole('button');

    fireEvent.mouseDown(fromSelect);
    const fromListbox = within(getByRole('listbox'));
    act(() => {
      fireEvent.click(fromListbox.queryByText('address2'));
    });
    expect(fromSelect).toHaveTextContent('address2');

    fireEvent.mouseDown(toSelect);
    const toListbox = within(getByRole('listbox'));
    act(() => {
      fireEvent.click(toListbox.queryByText('address3'));
    });
    expect(toSelect).toHaveTextContent('address3');
  });

  it('OrderForm select tariff', () => {
    const { getByTestId, getByRole, getAllByRole } = renderWithRedux(<OrderForm />, initialState);
    const [fromSelect, toSelect] = getAllByRole('button');

    fireEvent.mouseDown(fromSelect);
    const fromListbox = within(getByRole('listbox'));
    act(() => {
      fireEvent.click(fromListbox.queryByText('address1'));
    });
    fireEvent.mouseDown(toSelect);
    const toListbox = within(getByRole('listbox'));
    act(() => {
      fireEvent.click(toListbox.queryByText('address2'));
    });

    const tariff1 = getByTestId('tariff-1');
    const tariff2 = getByTestId('tariff-2');
    const tariff3 = getByTestId('tariff-3');

    fireEvent.click(tariff1);
    expect(tariff1).toHaveClass('tariffItem_active');
    expect(tariff2).not.toHaveClass('tariffItem_active');
    expect(tariff3).not.toHaveClass('tariffItem_active');

    fireEvent.click(tariff2);
    expect(tariff1).not.toHaveClass('tariffItem_active');
    expect(tariff2).toHaveClass('tariffItem_active');
    expect(tariff3).not.toHaveClass('tariffItem_active');

    fireEvent.click(tariff3);
    expect(tariff1).not.toHaveClass('tariffItem_active');
    expect(tariff2).not.toHaveClass('tariffItem_active');
    expect(tariff3).toHaveClass('tariffItem_active');
  });

  it('OrderForm render spinner', () => {
    initialState.order.isLoading = true;
    const { getByText } = renderWithRedux(<OrderForm />, initialState);

    expect(getByText(/spinner/i)).toBeInTheDocument();
  });

  it('OrderForm submit', () => {
    const { getAllByRole, getByRole } = renderWithRedux(<OrderForm />, initialState);
    const [fromSelect, toSelect, button] = getAllByRole('button');

    fireEvent.mouseDown(fromSelect);
    const fromListbox = within(getByRole('listbox'));
    act(() => {
      fireEvent.click(fromListbox.queryByText('address2'));
    });
    fireEvent.mouseDown(toSelect);
    const toListbox = within(getByRole('listbox'));
    act(() => {
      fireEvent.click(toListbox.queryByText('address3'));
    });

    const getRouteSpy = jest.spyOn(orderActions, 'getRoute');
    fireEvent.click(button);

    expect(getRouteSpy).toBeCalled();
  });

  it('OrderForm order success', () => {
    initialState.order.route = [
      [30.316273, 59.940578],
      [30.316589, 59.940495],
    ];

    const { getByText } = renderWithRedux(<OrderForm />, initialState);

    expect(getByText(/Заказ размещен/i)).toBeInTheDocument();
  });

  it('OrderForm reset form', () => {
    initialState.order.route = [
      [30.316273, 59.940578],
      [30.316589, 59.940495],
    ];
    const clearRouteSpy = jest.spyOn(orderActions, 'clearRoute');
    const { getByText, getByRole } = renderWithRedux(<OrderForm />, initialState);

    expect(getByText(/Заказ размещен/i)).toBeInTheDocument();
    fireEvent.click(getByRole('button'));

    expect(clearRouteSpy).toBeCalled();
    // todo Разобраться почему не меняется компонент
    // expect(getByText(/Заказ размещен/i)).not.toBeInTheDocument();
    // expect(getByText(/Заказать/i)).toBeInTheDocument();
  });

  it('OrderForm snapshot', () => {
    const orderForm = renderWithRedux(<OrderForm />, initialState);

    expect(orderForm).toMatchSnapshot();
  });
});
