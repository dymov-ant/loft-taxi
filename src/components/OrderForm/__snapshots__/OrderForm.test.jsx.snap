// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderForm component OrderForm snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="form"
      >
        <div
          class="form__top"
        >
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Откуда
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-from"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="mui-component-select-from"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                name="from"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled"
              data-shrink="false"
            >
              Куда
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-disabled="true"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-to"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled Mui-disabled"
                id="mui-component-select-to"
                role="button"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                name="to"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon Mui-disabled"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Заказать
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form"
    >
      <div
        class="form__top"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Откуда
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-component-select-from"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="mui-component-select-from"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              name="from"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled"
            data-shrink="false"
          >
            Куда
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
          >
            <div
              aria-disabled="true"
              aria-haspopup="listbox"
              aria-labelledby="mui-component-select-to"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled Mui-disabled"
              id="mui-component-select-to"
              role="button"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              name="to"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon Mui-disabled"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Заказать
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
